src/pages/ReportSettingsPage.js

```diff
@@ -107,23 +108,46 @@ class ReportSettingsPage extends Component {
             },
         };
 
+        this.roomNameInputRef = null;
+
         this.state = {
             newRoomName: this.props.report.reportName,
             errors: {},
         };
 
-        this.validateAndRenameReport = this.validateAndRenameReport.bind(this);
+        this.setRoomNameInputRef = this.setRoomNameInputRef.bind(this);
+        this.resetToPreviousName = this.resetToPreviousName.bind(this);
+        this.validateAndUpdatePolicyRoomName = this.validateAndUpdatePolicyRoomName.bind(this);
     }
 
-    validateAndRenameReport() {
-        if (!this.validate()) {
-            return;
+    /**
+     * Set the room name input ref
+     *
+     * @param {Element} el
+     */
+    setRoomNameInputRef(el) {
+        this.roomNameInputRef = el;
+    }
```

Comment:
Can we get rid of this function and just do it inline?

```js
<Component
    ref={el => this.roomNameInputRef = el}
```

Reply:
Yeah. That looks like the way we usually do it. I'll keep that in mind for the future. I was looking at ReportActionCompose which doesn't set the ref inline.